core/src/test/java/jenkins/util/JenkinsJVMTest.java

```diff
@@ -0,0 +1,36 @@
+package jenkins.util;
+
+import org.junit.Test;
+
+public class JenkinsJVMTest {
+
+    @Test
+    public void checkNotJenkinsJVM_WhenNotInAJenkinsJVM() {
+        JenkinsJVM.checkNotJenkinsJVM();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void checkJenkinsJVM_WhenNotInAJenkinsJVM() {
+        JenkinsJVM.checkJenkinsJVM();
+    }
+
+    @Test(expected = IllegalStateException.class)
+    public void checkNotJenkinsJVM_WhenInAJenkinsJVM() {
+        JenkinsJVM.setJenkinsJVM(true);
+        try {
+            JenkinsJVM.checkNotJenkinsJVM();
+        } finally {
+            JenkinsJVM.setJenkinsJVM(false);
```

Comment:
This is a "pure" unit test, so there is no `Jenkins` instance instantiated. The tests that have a `Jenkins` are the ones with `JenkinsRule` in the `test` module